### PR TITLE
possible different way to define target groups and listeners

### PR DESCRIPTION
**Pull request by yoshutch**
I haven't implemented it, this is just what examples could look like

### CODE REVIEW

**Comment by yoshutch** [comment]
I mainly want to know if this way is more intuitive and still flexible enough. Or do we need to rethink how to use the alb module altogether because its variables are too complex?

File: examples/blue-green/blue-green-example.tf [hunk 1]
```diff
@@ -8,55 +8,50 @@ module "acs" {
...
-        80,
-        443,
-        8001 // test listener
-      ]
-      name_suffix = "blue"
-      port        = 8000
-      config      = null // use default
+  listeners = {
+    80 = {
+      redirect_to = 443
```
Comment (yoshutch) [inline examples/blue-green/blue-green-example.tf]: might have to be an object like:
```
redirect_to = {
  host = null
  path = null
  port = 443
  protocol = "https"
}
```

**Comment by joshgubler** [comment]
Yes, I think this is more intuitive. What flexibility do you think we are losing?

**Comment by yoshutch** [comment]
I don't think we're losing flexibility, but one thought I had was to assume we're always going to have port 80 redirect to port 443, and have 443 always. But this probably wouldn't help with other configurations that we might need with this alb module so I decided to look at some other way to accomplish the redirecting listeners. Hence, this approach which I think is a bit better than before even.

**Pull request merged**